src/main/scala/firrtl/passes/RemoveCHIRRTL.scala

```diff
@@ -118,7 +119,7 @@ object RemoveCHIRRTL extends Pass {
         set_write(rws, "wdata", "wmask")
       val mem = DefMemory(s.info, s.name, s.tpe, s.size, 1, if (s.seq) 1 else 0,
                   rds map (_.name), wrs map (_.name), rws map (_.name))
-      Block(mem +: stmts)
+      if (stmts.nonEmpty) Block(mem +: stmts) else EmptyStmt
```

Comment:
@shunshou It's not clear to see whether or not it would break, but you can create no-port FIRRTL memories and they won't disappear. We could delete no-port memories in DCE later.